### DISCUSSION TITLE
Why is keycloak user in the root group when running docker container?

### DISCUSSION THREAD

**Discussion by Dreneg** [keycloak/keycloak#16641] [Q&A]
Background: In my project we are working on creating more secure docker images, and one step is changing our docker containers to run with non root user. We gladly noticed that keycloak does not use the root user by default, which is great. However the keycloak user is still in the root group (see: https://github.com/keycloak/keycloak/blob/main/quarkus/container/Dockerfile ). 

Question: We wonder if there are any specific reason why keycloak guys did not create a new group as well, instead of using root group.

Our currently "workaround" is just create our user and group and change the owner of the /opt/keycloak to our new user. Everything seems to be working, still we wonder if we missed something.

**Comment by datze** [comment]
@Dreneg did you ever since encounter problems with your own user?

**Reply by Dreneg** [reply]
@datze, No, so far we did not notice any problem with our custom user, except that our dockerfile a bit uglier than it should be. The relevant part is, where "my-keycloak" is a group and a user:
```
USER 0
RUN echo "my-keycloak:x:10001" >> /etc/group && \
    echo "my-keycloak:x:10001:10001:my-keycloak user:/opt/keycloak:/sbin/nologin" >> /etc/passwd

RUN chown -cR my-keycloak:my-keycloak /opt/keycloak
USER 10001
```

**Comment by sschu** [comment]
As I just learned, Keycloak runs with a random UID in OpenShift for security reasons. To still be able to prepare the file system, the GID is 0 so Keycloak needs to use this to access its file system when running in OpenShift.

**Reply by datze** [reply]
OK, that sounds plausible, but it also means that the admin group is not necessary if I know the UID. Right?

**Reply by sschu** [reply]
Correct.

**Reply by jonathancgt** [reply]
If Keycloak needs to rely on its process group for permissions, then why not just set the group owner to the non-root keycloak group for all resources it needs to access??
